distribution/lib/Standard/Table/0.0.0-dev/src/Internal/Fan_Out.enso

```diff
@@ -101,12 +103,26 @@ fan_out_to_rows_and_columns table input_column_id function column_names at_least
     input_storage = input_column.java_column.getStorage
     num_input_rows = input_storage.size
 
-    num_output_columns = column_names.length
+    # Create a column name map.
+    dynamic_columns = column_names.is_a Vector . not
```

Comment:
I think I don't understand this condition.

Looking at the type signature (the unchecked one), the `column_names` are always a Vector. So when is it possible that `column_names` _is not_ a Vector?

I think it would be good to clarify it somewhere.

Moreover, the documentation is beginning to be outdated - it specifies that `column_names` is supposed to be a vector - but I understand that now it can be something more? The documentation does not seem to cover the `dynamic_columns=True` case at all - so it is becoming outdated and may become misleading - I think we either have to update it or remove it (which would be sad because the doc here is really greatly written), but misleading doc may be worse than none at all.

Reply:
This will be easier following your suggestion above.